scripts/ci/sync_extensions.py

```diff
@@ -0,0 +1,146 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+# pylint: disable=broad-except
+
+import os
+import re
+import json
+from subprocess import check_output
+
+DEFAULT_TARGET_INDEX_URL = os.getenv('AZURE_EXTENSION_TARGET_INDEX_URL')
+STORAGE_ACCOUNT_KEY = os.getenv('AZURE_EXTENSION_TARGET_STORAGE_ACCOUNT_KEY')
+STORAGE_ACCOUNT = os.getenv('AZURE_EXTENSION_TARGET_STORAGE_ACCOUNT')
+STORAGE_CONTAINER = os.getenv('AZURE_EXTENSION_TARGET_STORAGE_CONTAINER')
+COMMIT_NUM = os.getenv('AZURE_EXTENSION_NUM') or 1
+
+
+def _get_updated_extension_names():
+    cmd = 'git --no-pager diff --diff-filter=ACMRT HEAD~{} -- src/index.json'.format(COMMIT_NUM)
+    updated_content = check_output(cmd.split()).decode('utf-8')
+    updated_urls = [line.replace('+', '') for line in updated_content.splitlines() if line.startswith('+') and not line.startswith('+++') and 'downloadUrl' in line]
+    updated_exts = []
+    NAME_REGEX = r'.*/([^/]*)-\d+.\d+.\d+'
+    if not updated_urls:
+        return updated_exts
+    for line in updated_urls:
+        search_result = re.search(r'"downloadUrl":\s+"(.*?)"', line)
```

Comment:
What if user manually update src/index.json and input `'downloadUrl'`?

Reply:
This also detects such case and is the same process as update by bot.

Reply:
I guess the re expression `r'"downloadUrl":\s+"(.*?)"` won't be ablt to search `'download'`

Reply:
no, it won't.